developers/prerequisites/OSGi.md

```diff
@@ -0,0 +1,130 @@
+
+The **OSGi runtime uses the information about the dependencies to *wire* the bundles and hides everything in this JAR unless it is explicitly exported**. The dependencies to the Java standard libraries are managed by the *Bundle-RequiredExecutionEnvironment* header, so it is not needed to import the Java core packages.
+
+A simple manifest file can have the following content:
+
+```xml
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: Example Plug-in
+Bundle-SymbolicName: com.example.myosgi; singleton:=true
+Bundle-Version: 1.0.0
+Bundle-RequiredExecutionEnvironment: JavaSE-1.7 
+Import-Package: org.example.required
+Export-Package: org.example.provided
+```
+[OSGi Service Platform Core Specification, Chapter 3.5][OSGi-Core] contains detailed information about the *Module Layer* and description of the headers (*Manifest-Version*, *Bundle-ManifestVersion*, *Bundle-Name*) used in this example.
+
+Bundles are used often to register and consume services. You will find more information about that in the [Services](#services) section.
+ 
+Lifecycle
+---------
+
+OSGi is a dynamic platform. That means that bundles may be *installed, uninstalled, started, stopped or updated* at runtime (See Table 1). The OSGi specification defines a mechanism how to manage the dependencies between the bundles and the functionality that they provide. This is achieved with the help of the *lifecycle* concept.
+
+The framework introduces a different states, transitions between these states and rules how this states are affecting the packages exported by the bundle and the services, that it provides. The table below shows the possible states of an OSGi bundle with a short explanation:
+
+Status | Description
+-------|---------
+INSTALLED | The bundle has been installed into the OSGi container, but some of the bundle's dependencies have not yet been met. The bundle requires packages that have not been exported by any currently installed bundle.
+RESOLVED | The bundle is installed and the OSGi system has connected up all the dependencies at a class level and made sure they are all resolved. The bundle can export the packages, that it provides.
+STARTING | A temporary state that the bundle goes through while the bundle is starting, after all dependencies have been resolved. The bundle is permitted to register services.
+ACTIVE | The bundle is running
+STOPPING | A temporary state that the bundle goes through while the bundle is stopping
+UNINSTALLED | The bundle has been removed from the OSGi container.  
+
+Table 1 (Source: <https://developer.atlassian.com/docs/atlassian-platform-common-components/plugin-framework/behind-the-scenes-in-the-plugin-framework/lifecycle-of-a-bundle>)
```

Comment:
I would hope that this definition is also part of the OSGi spec and thus you would not need to get info from atlassian.com (which could be a copyright issue). Could you either refer to osgi.org as the source or rewrite the descriptions in your own words and remove the source reference?